react/react2/week1/app/src/GithubContext.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { createContext, useState, useEffect } from "react";
+
+const GithubContext = createContext();
+
+const GithubProvider = ({ children }) => {
+  const [query, setQuery] = useState("");
+  const [users, setUsers] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
+
+  const getInput = (e) => {
+    const value = e.target.value;
+    setQuery(value);
+    if (value === "") {
+      setUsers([]);
+    }
+  };
+
+  useEffect(() => {
+    async function handleSubmit() {
+      if (query !== "") {
+        setLoading(true);
+        try {
+          const response = await fetch(
+            `https://api.github.com/search/users?q=${query}`
+          );
+          const data = await response.json();
+          setUsers(data.items);
+          setLoading(false);
+        } catch (error) {
+          setLoading(false);
+          setError(`💥 Error fetching: ${error.message}`);
+        }
```

Comment:
Though `setLoading()` works inside try() or catch(), it is suggested to use another method finally(), where you can do setLoading(). [for your reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch)